Sources/ServiceLifecycle/ServiceGroupConfiguration.swift

```diff
@@ -41,17 +44,142 @@ public struct ServiceGroupConfiguration: Hashable, Sendable {
+
+    /// The signals that lead to cancellation.
+    public var cancellationSignals = [UnixSignal]()
+
+    /// The group's logger.
+    public var logger: Logger
 
     /// The group's logging configuration.
-    public var logging: LoggingConfiguration
+    public var logging = LoggingConfiguration()
+
+
+    /// Initializes a new ``ServiceGroupConfiguration``.
+    /// 
+    /// - Parameters:
+    ///   - services: The groups's service configurations.
+    ///   - logger: The group's logger.
+    public init(
+        services: [ServiceConfiguration],
+        logger: Logger
+    ) {
+        self.services = services
+        self.logging = .init()
+        self.logger = logger
+    }
+
+    /// Initializes a new ``ServiceGroupConfiguration``.
+    ///
+    /// - Parameters:
+    ///   - services: The groups's service configurations.
+    ///   - gracefulShutdownSignals: The signals that lead to graceful shutdown.
+    ///   - cancellationSignals: The signals that lead to cancellation.
+    ///   - logger: The group's logger.
+    public init(
+        services: [ServiceConfiguration],
+        gracefulShutdownSignals: [UnixSignal] = .init(),
+        cancellationSignals: [UnixSignal] = .init(),
+        logger: Logger
+    ) {
+        self.services = services
+        self.logger = logger
+        self.gracefulShutdownSignals = gracefulShutdownSignals
+        self.cancellationSignals = cancellationSignals
+    }
 
     /// Initializes a new ``ServiceGroupConfiguration``.
     ///
-    /// - Parameter gracefulShutdownSignals: The signals that lead to graceful shutdown.
+    /// - Parameters:
+    ///   - services: The groups's services.
+    ///   - logger: The group's logger.
+    public init(
+        services: [Service],
+        logger: Logger
+    ) {
+        self.services = Array(services.map { ServiceConfiguration(service: $0) })
+        self.logger = logger
+    }
+
+    /// Initializes a new ``ServiceGroupConfiguration``.
+    ///
+    /// - Parameters:
+    ///   - services: The groups's services.
+    ///   - gracefulShutdownSignals: The signals that lead to graceful shutdown.
+    ///   - cancellationSignals: The signals that lead to cancellation.
+    ///   - logger: The group's logger.
+    public init(
+        services: [Service],
+        gracefulShutdownSignals: [UnixSignal] = .init(),
+        cancellationSignals: [UnixSignal] = .init(),
+        logger: Logger
+    ) {
+        self.services = Array(services.map { ServiceConfiguration(service: $0) })
+        self.logger = logger
+        self.gracefulShutdownSignals = gracefulShutdownSignals
+        self.cancellationSignals = cancellationSignals
+    }
+
+
+    @available(*, deprecated)
     public init(gracefulShutdownSignals: [UnixSignal]) {
+        self.services = []
         self.gracefulShutdownSignals = gracefulShutdownSignals
-        self.logging = .init()
+        self.logger = Logger(label: "Dummy logger") // This logger will never be used
```

Comment:
It _could_ be used I think? Would prefer setting the label to something obviously originating from here: `org.swift.service-lifecycle` or something.